src/snowflake/snowpark/mock/_stored_procedure.py

```diff
@@ -189,34 +296,49 @@ def _do_register_sp(
             anonymous,
         )
 
-        if is_pandas_udf:
-            raise TypeError("pandas stored procedure is not supported")
-
-        if packages or imports:
-            LocalTestOOBTelemetryService.get_instance().log_not_supported_error(
-                external_feature_name="uploading imports and packages for sprocs",
-                internal_feature_name="MockStoredProcedureRegistration._do_register_sp",
-                parameters_info={},
-                raise_error=NotImplementedError,
-            )
-
         check_python_runtime_version(self._session._runtime_version_from_requirement)
 
-        if udf_name in self._registry and not replace:
+        if sproc_name in self._registry and not replace:
             raise SnowparkSQLException(
-                f"002002 (42710): SQL compilation error: \nObject '{udf_name}' already exists.",
+                f"002002 (42710): SQL compilation error: \nObject '{sproc_name}' already exists.",
                 error_code="1304",
             )
 
+        if packages:
+            pass  # NO-OP
+
+        if imports is not None or type(func) is tuple:
+            self._sproc_level_imports[sproc_name] = set()
+
+        if imports is not None:
+            for _import in imports:
+                if type(_import) is str:
+                    self._import_file(_import, sproc_name=sproc_name)
+                else:
+                    local_path, import_path = _import
+                    self._import_file(local_path, import_path, sproc_name=sproc_name)
+
+        if type(func) is tuple:  # register from file
+            if sproc_name not in self._sproc_level_imports:
+                self._sproc_level_imports[sproc_name] = set()
+            module_name = self._import_file(func[0], sproc_name=sproc_name)
+            func = (module_name, func[1])
+
+        if sproc_name in self._sproc_level_imports:
+            sproc_imports = self._sproc_level_imports[sproc_name]
```

Comment:
I'm a bit curious about the `if sproc_name in self._sproc_level_imports:` logic here.
is this for re-registering a stored proc with the same name?

Reply:
Ditto.